kotlin/workflow-runtime/src/main/java/com/squareup/workflow/LaunchWorkflow.kt

```diff
@@ -112,6 +122,52 @@ fun <PropsT, OutputT : Any, RenderingT, RunnerT> launchWorkflowIn(
     beforeStart = beforeStart
 )
 
+@OptIn(ExperimentalCoroutinesApi::class, VeryExperimentalWorkflow::class)
+fun <PropsT, OutputT : Any, RenderingT> launchWorkflowIn2(
+  scope: CoroutineScope,
+  workflow: Workflow<PropsT, OutputT, RenderingT>,
+  props: StateFlow<PropsT>,
+  initialSnapshot: Snapshot? = null,
+  diagnosticListener: WorkflowDiagnosticListener? = null,
+  onOutput: suspend (OutputT) -> Unit
+): StateFlow<RenderingAndSnapshot<RenderingT>> {
```

Comment:
This is the main part I'd like early feedback on, the implementation details are still a bit of a WIP.